cmd/oras/root/manifest/index/update.go

```diff
@@ -0,0 +1,240 @@
+	tags            []string
+}
+
+func updateCmd() *cobra.Command {
+	var opts updateOptions
+	cmd := &cobra.Command{
+		Use:   "update <name>{:<tag>|@<digest>} {--add/--merge/--remove} {<tag>|<digest>} [...]",
+		Short: "[Experimental] Update and push an image index",
+		Long: `[Experimental] Update and push an image index. All manifests should be in the same repository
+		
+Example - remove a manifest and add two manifests from an index tagged 'v1'. The tag will point to the updated index:
+  oras manifest index update localhost:5000/hello:v1 --add linux-amd64 --add linux-arm64 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+Example - update an index by specifying its digest:
+  oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --remove linux-arm64
+
+Example - merge manifests from the index 'v1' to the index 'v2':
+  oras manifest index update localhost:5000/hello:v2 --merge v1
+
+Example - update an index and tag the updated index as 'v2.1.0' and 'latest':
+  oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --tag "v2.1.0" --tag "latest"
+  `,
+		Args: oerrors.CheckArgs(argument.AtLeast(1), "the destination index to update"),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.RawReference = args[0]
+			return option.Parse(cmd, &opts)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return updateIndex(cmd, opts)
+		},
+	}
+	option.ApplyFlags(&opts, cmd.Flags())
+	cmd.Flags().StringArrayVarP(&opts.addArguments, "add", "", nil, "add manifests to the index")
+	cmd.Flags().StringArrayVarP(&opts.mergeArguments, "merge", "", nil, "merge the manifests of another index")
+	cmd.Flags().StringArrayVarP(&opts.removeArguments, "remove", "", nil, "manifests to remove from the index")
+	cmd.Flags().StringArrayVarP(&opts.tags, "tag", "", nil, "tags for the updated index")
+	return oerrors.Command(cmd, &opts.Target)
+}
+
+func updateIndex(cmd *cobra.Command, opts updateOptions) error {
+	// if no update flag is used, do nothing
+	if !cmd.Flags().Changed("add") && !cmd.Flags().Changed("remove") && !cmd.Flags().Changed("merge") {
+		opts.Println("No update flag is used. There's nothing to update.")
+		return nil
+	}
+	ctx, logger := command.GetLogger(cmd, &opts.Common)
+	target, err := opts.NewTarget(opts.Common, logger)
+	if err != nil {
+		return err
+	}
+	if err := opts.EnsureReferenceNotEmpty(cmd, true); err != nil {
+		return err
+	}
+	index, err := fetchIndex(ctx, target, opts)
+	if err != nil {
+		return err
+	}
+	manifests, err := addManifests(ctx, index.Manifests, target, opts)
+	if err != nil {
+		return err
+	}
+	manifests, err = mergeIndexes(ctx, manifests, target, opts)
+	if err != nil {
+		return err
+	}
+	manifests, err = removeManifests(ctx, manifests, target, opts)
+	if err != nil {
+		return err
+	}
```

Comment:
Does the order of updates matter?

Reply:
Per off line discussion, `oras` does not guarantee the order if multiple update commands (e.g. `add`, `remove` used together) are used. The order is preserved if only one type of command is used.